editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-element-renderer-utils.tsx

```diff
@@ -345,8 +346,20 @@ export function renderCoreElement(
   }
 }
 
-export function filterJSXElementChildIsTextOrNewline(c: JSXElementChild): c is JSXTextBlock {
-  return c.type === 'JSX_TEXT_BLOCK' || (c.type === 'JSX_ELEMENT' && c.name.baseVariable === 'br')
+export function textOrNullFromJSXElement(c: JSXElementChild): string | null {
+  switch (c.type) {
+    case 'JSX_TEXT_BLOCK':
+      return c.text
+    case 'JSX_ELEMENT':
+      return c.name.baseVariable === 'br' ? '\n' : null
+    case 'JSX_ARBITRARY_BLOCK':
+      if (c.transpiledJavascript === `return ${c.javascript}`) {
+        return `{${c.originalJavascript}}`
+      }
+      return null
+    default:
+      return null
```

Comment:
minor, this should be `assertNeven(c)` instead so it'll be easier to extend the type (when it comes to that)